decavision/model_testing/testing.py

```diff
@@ -253,3 +256,198 @@ def generate_classification_report(self, path):
+        for image_path in images:
+            # prepare the image
+            image_tensor = data_utils.prepare_image(image_path, self.input_shape, self.rescaling)
+            # make and decode the prediction
+            result = self.model.predict(image_tensor)[0]
+            # print image and top predictions
+            ##top_pred = np.argsort(result)[::-1][:5] # GET PREDS > THRESHOLD (0.50) KEEP IT AS A USER ARGUMENT. 
+            top_pred = result > threshold
+            # Name of the true class.
+            cls_pred_name = np.array(categories)[top_pred]
+            cls_pred_perc = result[top_pred] * 100
+            cls_true = self.values[os.path.basename(image_path)]
+            if plot:
+                fig, ax = plt.subplots()
+                if self.model.name in ["Inception", "Xception"]:
+                    ax.imshow(image_tensor[0], interpolation='nearest')
+                else:
+                    ax.imshow(image_tensor[0].astype('uint8'), interpolation='nearest')
+                xlabel = 'Prediction :\n'
+                for (x, y) in zip(cls_pred_name, cls_pred_perc):
+                    xlabel += '{0}, {1:.2f}%\n'.format(x, y)
+                ax.set_xlabel(xlabel)
+                ax.set_xticks([])
+                ax.set_yticks([]) 
+                plt.tight_layout() 
+                plt.show()
+                fig.savefig("predicated_images/" + os.path.basename(image_path))                
+            else:
+                print('\nImage: ', image_path)
+                print("True label:", cls_true)
+                for i in range(len(cls_pred_perc)):
+                    print('Prediction: {} (probability {}%)'.format(cls_pred_name[i], round(cls_pred_perc[i])))
+
+    def evaluate(self, path):
+        """
+        Calculate the f1-score of the model on a dataset of images. The images must be
+        in single folder for each class.
+
+        Arguments:
+            path (str): location of the dataset
+        """
+        
+        generator = self._load_dataset(path)
+        results = self.model.evaluate(generator)
+        print('f1-score of', round(results[-1] * 100,3), '%')
+        
+    def generate_metrics(self, path, threshold=0.5):
+        """
+        Computes classification report and confusion matrix resulting from predictions on 
+        a dataset of images and prints the results. Images must be located in a single folder 
+        for each class. 
+        
+        Arguments:
+            path (str): location of the images
+            threshold (int): threshold for prediction (default to 0.5)
+        """
+    
+        # getting list of true and predicted labels
+        generator = self._load_dataset(path)
+        cls_true = generator.classes #true label for each image
+        labels = list(generator.class_indices.keys()) # list of total classes
+        cls_pred = self.model.predict(generator)
+        cls_pred = cls_pred > threshold
```

Comment:
Isn't this already in the format that we want? And for the true labels they are also given by the generator in the right format. You just need to extract them and combine them in a single array. For example you could do something like
```
y_true = []
for i in range(generator.n):
    y_true.append(generator.next()[1])
y_true = np.array(y_true)
```